File: back2/api/v1/models/codigo_confirmacion.py
from sqlalchemy import Column, BigInteger, String, DateTime
from db.base import Base

class CodigoConfirmacion(Base):
    __tablename__ = "codigosconfirmacion"
    id = Column(BigInteger, primary_key=True, autoincrement=True, index=True)
    email = Column(String(255), nullable=False, unique=True, index=True)
    codigo = Column(String(255), nullable=False, unique=True, index=True)
    vencimiento = Column("codigo_verificacion_expira", DateTime, nullable=True)

File: back/src/main/java/com/example/ss1/models/CodigoConfirmacion.java
package com.example.ss1.models;

import java.time.LocalDateTime;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;

@Entity
@Table(name = "codigosconfirmacion")
public class CodigoConfirmacion {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String email;
    @Column(nullable = false)
    private String codigo;
    @Column(nullable = false)
    private LocalDateTime vencimiento;
    @Column(nullable = false)
    private int tipo;
    
    public Long getId() {
        return id;
    }
    public void setId(Long id) {
        this.id = id;
    }
    public String getEmail() {
        return email;
    }
    public void setEmail(String email) {
        this.email = email;
    }
    public String getCodigo() {
        return codigo;
    }
    public void setCodigo(String codigo) {
        this.codigo = codigo;
    }
    public LocalDateTime getVencimiento() {
        return vencimiento;
    }
    public void setVencimiento(LocalDateTime vencimiento) {
        this.vencimiento = vencimiento;
    }
    public int getTipo() {
        return tipo;
    }
    public void setTipo(int tipo) {
        this.tipo = tipo;
    }

    

}
